Domain

maciejka opened the pull request.

maciejka committed:
Nile

maciejka committed:
Storage vars

maciejka committed:
First lines of frob

maciejka committed:
Update to Cairo 0.7.0

maciejka committed:
Add assert_le

maciejka committed:
frob draft

maciejka committed:
fork draft

maciejka committed:
grab draft

maciejka committed:
heal, suck, fold drafts

maciejka committed:
move draft

maciejka committed:
flux draft

maciejka committed:
slip draft

maciejka committed:
cage draft

maciejka committed:
constructor

maciejka committed:
Smaller functions

maciejka committed:
Synced with xdomain-dss

maciejka committed:
Helper functions extracted to a separate files

nulven committed:
Add test contracts

maciejka committed:
Safe math

maciejka committed:
Add nonnegative checks to signed ops

nulven committed:
Merge branch 'draft' of github.com:makerdao/starknet-dss into draft

maciejka committed:
Implement assertions

nulven committed:
Add protostar

maciejka committed:
Add Uint256 checks

maciejka committed:
Add todos

maciejka committed:
DaiJoin and GemJoin TODOs

maciejka committed:
Add comment

nulven committed:
Add python tests

nulven committed:
Add account

nulven committed:
Update gitignore

nulven committed:
Finish fold test

nulven committed:
Finish vat tests

maciejka committed:
Move event declations up in dai_join

maciejka committed:
Pin dill version

maciejka committed:
Remove unncessary files

maciejka committed:
Test fixes and comments

maciejka committed:
Setup CI

nulven committed:
Initial reorganization

nulven committed:
Fix tests

nulven committed:
More adjustments

nulven committed:
Fix CI

nulven committed:
Add tests vat directory

maciejka committed:
Make few function declarations more compact

maciejka committed:
Add Line constants

maciejka committed:
Merge pull request #1 from makerdao/nu/vat-tests

Nu/vat tests

maciejka committed:
Cairo == 0.9.1

maciejka committed:
swell

maciejka committed:
safe_math fixes, swell checks

maciejka committed:
missing events

maciejka committed:
missing views

maciejka committed:
dai_join

maciejka committed:
join tests

maciejka committed:
join links to solidity version

maciejka committed:
frob 'based on' link

maciejka committed:
fork 'based on' link

maciejka committed:
fold 'based on' link

maciejka committed:
remove bite

maciejka committed:
dai_join 'based on' link

EvolveArt committed:
feat: add amarna to CI ✨

EvolveArt committed:
Add poetry and pytest-xdist for test parallelization (#3)

* feat: add pytest-xdist 🚀

* feat: add poetry ✨

maciejka committed:
gem_join clean up

EvolveArt committed:
feat: dai contract and tests ✨ (#4)

- add `test_` prefix for pytest
- refractor dai tests using new `conftest`
- add `check_event` and `check_balances` in `conftest`

maciejka committed:
teleport join placeholders

maciejka committed:
validate signatures

maciejka committed:
request_mint, in progress

nulven committed:
Add teleport router draft

maciejka committed:
add request_mint

maciejka committed:
fixes

maciejka committed:
user -> usr

EvolveArt committed:
Evolve/add hardhat (#11)

* feat: add hardhat ✨

* Add hardhat tests (#7)

* teleport join placeholders

* validate signatures

* request_mint, in progress

* Add teleport router draft

* Fix input name

* Add hardhat tests

* fix: ci

Co-authored-by: Maciek Kamiński <aundumla@gmail.com>
Co-authored-by: Evolve <Artevolve@yahoo.com>

Co-authored-by: Nick Ulven <nickulven@gmail.com>
Co-authored-by: Maciek Kamiński <aundumla@gmail.com>

EvolveArt committed:
Squashed commit of the following:

commit d2266683a1d5eb71e035095034b47df1b42b18c6
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Sep 5 13:29:39 2022 +0200

    feat: rebase test folder

commit 7d6bba6eeb79148082fdd77c256e325c605fd6d4
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Sep 5 13:18:54 2022 +0200

    fix: test folder

commit b1e2f3af1c27781cc187b6eb886e5a540454ea8d
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Sep 5 13:10:56 2022 +0200

    fix: gitignore

commit b600fd48dd8cfe8c0eeb3c2c6445dbfd2da13694
Author: Evolve <Artevolve@yahoo.com>
Date:   Thu Aug 11 20:46:15 2022 +0200

    feat: teleport_join (no tests) ✨

commit 347589bc5d1c87a8b5c08e216fa1e121848c6acc
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Aug 9 15:46:09 2022 +0200

    feat: teleport_join (wip) ✨

commit 9de0f9eee55ae8b533129c8f3684f17a41377431
Author: Maciek Kamiński <aundumla@gmail.com>
Date:   Wed Aug 3 13:51:05 2022 +0200

    teleport join placeholders

EvolveArt committed:
fix: compile errors 🔨

EvolveArt committed:
Update .gitignore

EvolveArt committed:
Squashed commit of the following:

commit 9d053d2b27ed536d987e98cbd2aa290849108a76
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Aug 30 14:53:26 2022 +0400

    feat: teleport_constant_fee test ✨

commit 9669f1904986721b42d0434aa7803465667e6ed0
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Aug 30 14:22:33 2022 +0400

    feat: teleport_constant_fee ✨

EvolveArt committed:
fix: amount type compile 🔨

maciejka committed:
Update comments

EvolveArt committed:
feat: helpers ✨

- wad/rad/ray/uint helpers
- invoke helper
- move checkAuth in utils

maciejka commented on contracts/starknet/safe_math.cairo:
```diff
@@ -0,0 +1,174 @@
+    return (res)
+end
+
+func div_rem{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```
I would call this function just `div` and do not return reminder. It is just an equivalent of solidity `/`. @EvolveArt

EvolveArt replied:
well we still get the remainder returned in this case. We could not return it and call it simply `div` then

maciejka replied:
We are not interested in remainder. This particular division should round down towards zero.

EvolveArt replied:
right, renamed it to div and removed the remainder

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,938 @@
+    func getFee(
+        calldata : TeleportGUID,
+        line : Uint256,
+        debt : Uint256,
```
`Int256`? @EvolveArt

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,938 @@
+#     function slip(bytes32, address, int256) external;
+# }
+@contract_interface
+namespace VatLike:
```
Order of methods should be the same as in the solidity comments above. @EvolveArt

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,938 @@
+    return ()
+end
+
+# function file(bytes32 what, address data) external auth {
```
Left not implemented for later? @EvolveArt

maciejka commented on contracts/starknet/safe_math.cairo:
```diff
@@ -0,0 +1,174 @@
+# function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+#         z = x <= y ? x : y;
+#     }
+func _min{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```
Why not call it just `min`? `_` prefix is reserved to functions with mixed uint/int params. @EvolveArt

EvolveArt replied:
Was just using the same name as the solidity one there but indeed makes sense

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,938 @@
+    let (minus_debt) = uint256_neg(debt_)
+    let (mint_needed) = le_int(minus_debt, amt_to_take)
+    let (should_mint) = either(debt_pos, mint_needed)
+    if should_mint == 1:
```
Not sure if it is worth the complexity, `should_mint` can be optimized via lazy evaluation.

EvolveArt replied:
how would you go about it in this case ? will try smth

maciejka replied:
I would leave a TODO. Not worth the complexity.

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+    what : felt, domain : felt, data : felt
+):
+    # address prevGateway = gateways[domain];
+    let prev_gateway = _gateways.read(domain)
```
I would comment this and following lines with prev_gatevay out as it is unfinished code.

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+    end
+
+    # gateways[domain] = data;
+    _gateways.write(domain, data)
```
In next `file` function validation precedes storage write. Let's keep it consistent event if solidity version is not.

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+    # gateways[domain] = data;
+    _gateways.write(domain, data)
+
+    # else revert("TeleportRouter/file-unrecognized-param");
```
Comment seems to be incomplete.

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+
+    # if (gateway == address(0)) gateway = parent;
+    if gateway == 0:
+        _gateway.write(parent)
```
Storage should not be updated here. See settle method below.

EvolveArt replied:
replaced by a `get_gateway` function for conditional assignment it's the best way I know of

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+
+    # require(msg.sender == parent || msg.sender == gateways[sourceDomain], "TeleportRouter/sender-not-gateway");
+    with_attr error_message("TeleportRouter/sender-not-gateway"):
+        let (parent_eq) = is_not_zero(caller - parent)
```
It should be `is_zero` here?

maciejka commented on contracts/starknet/teleport_router.cairo:
```diff
@@ -0,0 +1,297 @@
+    with_attr error_message("TeleportRouter/sender-not-gateway"):
+        let (parent_eq) = is_not_zero(caller - parent)
+        let (source_gateway) = _gateways.read(source_domain)
+        let (gateway_eq) = is_not_zero(caller - source_gateway)
```
It should be `is_zero` here?

maciejka committed:
Merge domain branch

maciejka committed:
Merged

EvolveArt committed:
feat: vat tests (wip) ✨

- add helper functions frob/fold/suck/hope
- add art view function

EvolveArt committed:
fix: code review changes 🔨

EvolveArt committed:
feat: migrate cairo 0.10.0 ✨ (#18)

maciejka committed:
Merge remote-tracking branch 'origin/domain' into teleport-oracle-auth

maciejka committed:
Convert teleport_oracle_auth to 0.10.0

maciejka committed:
Remove outdated files

EvolveArt committed:
feat: fork tests ✨

EvolveArt committed:
feat: grab/heal/suck tests ✨

maciejka committed:
GUID hash

maciejka committed:
Update contracts/starknet/teleport_oracle_auth.cairo

Co-authored-by: Evolve <Artevolve@yahoo.com>

maciejka committed:
Update contracts/starknet/teleport_oracle_auth.cairo

Co-authored-by: Evolve <Artevolve@yahoo.com>

maciejka committed:
improve signature validation

maciejka committed:
enforce signers to be unique

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,934 @@
+    let (amt_to_take: Uint256) = min(pending, under_ceiling);
+
+    // uint256 fee = vatLive ? FeesLike(fees[teleportGUID.sourceDomain]).getFee(teleportGUID, line_, debt_, pending, amtToTake) : 0;
+    //         require(fee <= maxFeePercentage * amtToTake / WAD, "TeleportJoin/max-fee-exceed");
```
Max fee assertion logic seems to be missing in Cairo version.

maciejka replied:
@EvolveArt

maciejka commented on contracts/starknet/teleport_join.cairo:
```diff
@@ -0,0 +1,934 @@
+
+    let (debt_pos) = _ge_0(debt_);
+    let (minus_debt) = uint256_neg(debt_);
+    let (mint_needed) = le_int(minus_debt, amt_to_take);
```
Solidity version is more nuanced. It will not calculate `uint256(-debt_) < amtToTake` if `debt_ >= 0`. Which means  uint256_neg(debt_) is > 0 and `le_int` in line 489 should be an unsigned comparison. By the way in Solidity it is `<`not `<=`.

maciejka replied:
@EvolveArt

maciejka commented on contracts/starknet/assertions.cairo:
```diff
@@ -0,0 +1,123 @@
+}
+
+// signed!
+func _ge_0{range_check_ptr}(a: Uint256) -> (res: felt) {
```
It should be `a: Int256`

maciejka commented on contracts/starknet/assertions.cairo:
```diff
@@ -0,0 +1,123 @@
+}
+
+func le{range_check_ptr}(a: Uint256, b: Uint256) -> (res: felt) {
+    alloc_locals;
```
There is an `uint256_le` in the std lib now.

maciejka replied:
My intention was to use `uint256_le` in implementation of `le` not to replace `le` with `uint256_le`.

maciejka commented on contracts/starknet/assertions.cairo:
```diff
@@ -0,0 +1,123 @@
+}
+
+func le_int{range_check_ptr}(a: Int256, b: Int256) -> (res: felt) {
+    alloc_locals;
```
Sth is wrong here. Why not just use `uint256_signed_le`?

maciejka replied:
This function can be named `_le` to be consistent with e.g.: `_ge_0`

maciejka commented on contracts/starknet/assertions.cairo:
```diff
@@ -0,0 +1,123 @@
+}
+
+// signed!
+func _le_0{range_check_ptr}(a: Uint256) -> (res: felt) {
```
`a: Uint256` -> `a: Int256`

maciejka committed:
add comment

maciejka committed:
change return type

maciejka committed:
Merge pull request #8 from makerdao/teleport-oracle-auth

Teleport oracle auth

maciejka committed:
fix fold, suggested by Veridise

maciejka committed:
Merge pull request #23 from makerdao/mk/fold-fix

Fold fix

maciejka committed:
remove outdated python tests (#24)

EvolveArt committed:
Teleport Join Tests ✨ (#19)

* feat: add mocks & test init (wip) ✨

* fix: code review fixes 🔨

* feat: file functions + new tests ✨

* feat: more tests + cure function ✨

* feat: more tests + join fixes ✨

* feat: last tests ✨

* fix: fixing tests 🔨

* fix: more tests fixes 🔨

* fix: more test fixes 🔨

- added `mock_vat`
- update hardhat plugin

* fix: more test fixes 🔨

- remove bytes32 conversion
- temp remove `daiSent` & `totalFees` checks

* fix: more test fixes 🔨

- write dai address at initialization

* fix: more test fixes 🔨

* fix: minor test fixes 🔨

- add signed/signed sub function (from warp)
- fix bug in `vat_mock`

* fix: teleport join tests ✨

- todo: handle negative integers in js

* fix: ci ♻️

* fix: ci ♻️

* fix: ci ♻️

* fix: ci ♻️

* fix: ci ♻️

* fix: ci ♻️

* fix: type name ✍️

* feat: solidity comments ✍️

* feat: teleport join changes ✨

- update code to follow https://github.com/makerdao/dss-teleport/commits/routing-v2/src/TeleportJoin.sol

* fix: remove python test files 🔨

* fix: review changes ✍️

* fix: test solidity names ✍️

* fix: switch to docker env ♻️

EvolveArt committed:
feat: return values w/ transaction trace ✨ (#31)

EvolveArt committed:
Teleport Router tests ✨ (#22)

* feat: teleport router tests ✨

* feat: solidity comments + new changes ✨

- Update code to follow https://github.com/makerdao/dss-teleport/blob/routing-v2/src/TeleportRouter.sol
- add solidity comments for easier review

* fix: ci ♻️

* fix: ci ♻️

* Update pyproject.toml

* fix: ci ♻️

* fix: solidity comments ✍️

* fix: typo ✍️

* fix: test comments indentation ✍️

* Update hardhat.config.ts

* fix: switch to docker env ♻️

* fix: review ♻️

* fix: test names comments ✍️

* Update chain-l2.sh

EvolveArt committed:
Teleport Oracle Auth tests ✨ (#21)

* feat: teleport oracle auth tests ✨ (wip)

- todo: fix signatures bug

* fix: ci ♻️

* feat: solidity comments ✨

EvolveArt committed:
Evolve/dai tests (#13)

* feat: dai tests (wip) ✨

* feat: dai tests (wip) ✨

- Note that user1 denotes self, user2 = user2 and admin = user1

* feat: dai tests (finished) ✨

* feat: update starknet plugin 🔨

* feat: versions migration ✨

* fix: ci ♻️

* Delete poetry.lock

* fix: ci ♻️

* fix: solidity comments ✍️

* fix: typo ✍️

* fix: review ✍️

EvolveArt committed:
fix: compile error 🔨

EvolveArt committed:
Evolve/cure end (#15)

* feat: cure rewrite ✨

- simple rewrite of cure contract

* fix: move teleports contracts ✍️

* feat: cure tests ✨ (wip)

* feat: cure tests ✨

- issue w/ starknet-devnet time increase

* feat: end rewrite ✨ (wip)

- Added wad/ray library from Aave bridge repo

* feat: end rewrite ✨

- only remains `file` functions
- should we use `Rad/Wad` types ?

* feat: end tests ✨ (wip)

- should everything be tested strictly like L1 contracts since Jug/Pot/Spot are not present on L2

* feat: end Cage_ilk 🔨

* fix: update storage vars 🔨

* fix: update hardhat + fix cure wait test 🔨

* fix: dump file path ✍️

* Update hardhat.config.ts

* Update tests.yml

* Update package.json

* Update pyproject.toml

* fix: devnet version ✍️

* fix: ci 🔨

* fix: gitignore ✍️

* feat: versions migration ✨

* fix: solidity comments ✍️ (cure tests)

* Squashed commit of the following:

commit 78251e6010bf1de6cf06afd40dcc16edb313501f
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Oct 18 14:03:57 2022 +0200

    Teleport Oracle Auth tests ✨ (#21)

    * feat: teleport oracle auth tests ✨ (wip)

    - todo: fix signatures bug

    * fix: ci ♻️

    * feat: solidity comments ✨

commit e85096b28556bc75ddae7e3dbf14e9aee8f6631f
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Oct 18 14:03:38 2022 +0200

    Teleport Router tests ✨ (#22)

    * feat: teleport router tests ✨

    * feat: solidity comments + new changes ✨

    - Update code to follow https://github.com/makerdao/dss-teleport/blob/routing-v2/src/TeleportRouter.sol
    - add solidity comments for easier review

    * fix: ci ♻️

    * fix: ci ♻️

    * Update pyproject.toml

    * fix: ci ♻️

    * fix: solidity comments ✍️

    * fix: typo ✍️

    * fix: test comments indentation ✍️

    * Update hardhat.config.ts

    * fix: switch to docker env ♻️

    * fix: review ♻️

    * fix: test names comments ✍️

    * Update chain-l2.sh

commit db0b50ce0ee725571ed71c7a38faa74eb291e5ba
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Oct 17 13:59:00 2022 +0200

    feat: return values w/ transaction trace ✨ (#31)

commit f0298320045fe1612d320286bed2ee4c7db53602
Author: Evolve <Artevolve@yahoo.com>
Date:   Thu Oct 13 14:23:28 2022 +0200

    Teleport Join Tests ✨ (#19)

    * feat: add mocks & test init (wip) ✨

    * fix: code review fixes 🔨

    * feat: file functions + new tests ✨

    * feat: more tests + cure function ✨

    * feat: more tests + join fixes ✨

    * feat: last tests ✨

    * fix: fixing tests 🔨

    * fix: more tests fixes 🔨

    * fix: more test fixes 🔨

    - added `mock_vat`
    - update hardhat plugin

    * fix: more test fixes 🔨

    - remove bytes32 conversion
    - temp remove `daiSent` & `totalFees` checks

    * fix: more test fixes 🔨

    - write dai address at initialization

    * fix: more test fixes 🔨

    * fix: minor test fixes 🔨

    - add signed/signed sub function (from warp)
    - fix bug in `vat_mock`

    * fix: teleport join tests ✨

    - todo: handle negative integers in js

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: type name ✍️

    * feat: solidity comments ✍️

    * feat: teleport join changes ✨

    - update code to follow https://github.com/makerdao/dss-teleport/commits/routing-v2/src/TeleportJoin.sol

    * fix: remove python test files 🔨

    * fix: review changes ✍️

    * fix: test solidity names ✍️

    * fix: switch to docker env ♻️

commit 004ba68e3677ee3670be882b362567f4f92023bc
Author: Maciej Kamiński <aundumla@gmail.com>
Date:   Wed Oct 12 12:54:56 2022 +0200

    remove outdated python tests (#24)

commit edc15958c77ffdf54d8d894c8d7dded6964ea82a
Merge: 5268b5b 8713f85
Author: Maciej Kamiński <aundumla@gmail.com>
Date:   Tue Oct 11 16:47:24 2022 +0200

    Merge pull request #23 from makerdao/mk/fold-fix

    Fold fix

commit 8713f857849cc7a2a576d77ccd441d8b6cff1786
Author: Maciek Kamiński <aundumla@gmail.com>
Date:   Tue Oct 11 15:03:54 2022 +0200

    fix fold, suggested by Veridise

* fix: minor fixes 🔨

* fix: cure tests 🧪

* feat: end tests (wip) 🔨

* feat: end file functions 🔨

* feat: end tests (wip) 🧪

* feat: more end tests 🧪

* feat: end tests 🧪

- first draft finished, now mostly bug fixes

* feat: mock contracts 🧪

* fix: migrate 0.10.1 ⬆️

* Squashed commit of the following:

commit b4487b67bbb7161bda9b7c6c145731f638c41b81
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Oct 24 17:47:07 2022 +0200

    fix: compile error 🔨

commit 2fdafa25ad764fee380855e2cb32350b8f4e53de
Author: Evolve <Artevolve@yahoo.com>
Date:   Thu Oct 20 17:29:26 2022 +0200

    Evolve/dai tests (#13)

    * feat: dai tests (wip) ✨

    * feat: dai tests (wip) ✨

    - Note that user1 denotes self, user2 = user2 and admin = user1

    * feat: dai tests (finished) ✨

    * feat: update starknet plugin 🔨

    * feat: versions migration ✨

    * fix: ci ♻️

    * Delete poetry.lock

    * fix: ci ♻️

    * fix: solidity comments ✍️

    * fix: typo ✍️

    * fix: review ✍️

commit 78251e6010bf1de6cf06afd40dcc16edb313501f
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Oct 18 14:03:57 2022 +0200

    Teleport Oracle Auth tests ✨ (#21)

    * feat: teleport oracle auth tests ✨ (wip)

    - todo: fix signatures bug

    * fix: ci ♻️

    * feat: solidity comments ✨

commit e85096b28556bc75ddae7e3dbf14e9aee8f6631f
Author: Evolve <Artevolve@yahoo.com>
Date:   Tue Oct 18 14:03:38 2022 +0200

    Teleport Router tests ✨ (#22)

    * feat: teleport router tests ✨

    * feat: solidity comments + new changes ✨

    - Update code to follow https://github.com/makerdao/dss-teleport/blob/routing-v2/src/TeleportRouter.sol
    - add solidity comments for easier review

    * fix: ci ♻️

    * fix: ci ♻️

    * Update pyproject.toml

    * fix: ci ♻️

    * fix: solidity comments ✍️

    * fix: typo ✍️

    * fix: test comments indentation ✍️

    * Update hardhat.config.ts

    * fix: switch to docker env ♻️

    * fix: review ♻️

    * fix: test names comments ✍️

    * Update chain-l2.sh

commit db0b50ce0ee725571ed71c7a38faa74eb291e5ba
Author: Evolve <Artevolve@yahoo.com>
Date:   Mon Oct 17 13:59:00 2022 +0200

    feat: return values w/ transaction trace ✨ (#31)

commit f0298320045fe1612d320286bed2ee4c7db53602
Author: Evolve <Artevolve@yahoo.com>
Date:   Thu Oct 13 14:23:28 2022 +0200

    Teleport Join Tests ✨ (#19)

    * feat: add mocks & test init (wip) ✨

    * fix: code review fixes 🔨

    * feat: file functions + new tests ✨

    * feat: more tests + cure function ✨

    * feat: more tests + join fixes ✨

    * feat: last tests ✨

    * fix: fixing tests 🔨

    * fix: more tests fixes 🔨

    * fix: more test fixes 🔨

    - added `mock_vat`
    - update hardhat plugin

    * fix: more test fixes 🔨

    - remove bytes32 conversion
    - temp remove `daiSent` & `totalFees` checks

    * fix: more test fixes 🔨

    - write dai address at initialization

    * fix: more test fixes 🔨

    * fix: minor test fixes 🔨

    - add signed/signed sub function (from warp)
    - fix bug in `vat_mock`

    * fix: teleport join tests ✨

    - todo: handle negative integers in js

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: ci ♻️

    * fix: type name ✍️

    * feat: solidity comments ✍️

    * feat: teleport join changes ✨

    - update code to follow https://github.com/makerdao/dss-teleport/commits/routing-v2/src/TeleportJoin.sol

    * fix: remove python test files 🔨

    * fix: review changes ✍️

    * fix: test solidity names ✍️

    * fix: switch to docker env ♻️

commit 004ba68e3677ee3670be882b362567f4f92023bc
Author: Maciej Kamiński <aundumla@gmail.com>
Date:   Wed Oct 12 12:54:56 2022 +0200

    remove outdated python tests (#24)

commit edc15958c77ffdf54d8d894c8d7dded6964ea82a
Merge: 5268b5b 8713f85
Author: Maciej Kamiński <aundumla@gmail.com>
Date:   Tue Oct 11 16:47:24 2022 +0200

    Merge pull request #23 from makerdao/mk/fold-fix

    Fold fix

commit 8713f857849cc7a2a576d77ccd441d8b6cff1786
Author: Maciek Kamiński <aundumla@gmail.com>
Date:   Tue Oct 11 15:03:54 2022 +0200

    fix fold, suggested by Veridise

* fix: compile errors 🔨

* fix: env ✍️

* fix: test fixes 🧪

* fix: review changes 🤝

* fix: replace user by usr ✍️

* fix: debug CI (wip) 🔨

* fix: vat tests 🧪 (wip)

* fix: vat tests 🧪 (wip)

* fix: review changes 🤝

* fix: review changes 🤝

* fix: negative numbers vat 🧪

* fix: negative numbers + tests 🔨

* fix: oracle auth tests 🧪

* feat: spotter ✨

* feat: pot ✨

* fix: revert binding 🔨

* fix: review changes 🤝

* fix: rename users + end file ref ✍️

* fix: remove sleeps 🔨

* fix: rad/ray utils 🧪

* fix: remove ray/wad utils 🔨

* fix: overflow bounds ✍️

* fix: end tests 🧪

* fix: CI 🎉

* fix: CI 🎉

* fix: port ✍️

* fix: typo ✍️

EvolveArt committed:
fix: delete pot/spotter 🔨

EvolveArt committed:
Evolve/split repos (#41)

* fix: delete teleport related files 👏

* fix: remove unused mocks 🔨

* Update tests.yml

* Revert "Update tests.yml"

This reverts commit 5dbce91b2d2f55750f5c1fd912445e908acdddbb.

EvolveArt committed:
Add Spotter / Pot / Jug (#40)

* Revert "fix: delete pot/spotter 🔨"

This reverts commit 5f2470a969fc9eb614eac3d09003570c9e5be9c1.

* feat: jug ✨

* feat: pot tests 🧪

* fix: pot test fixes ✍️

* feat: jug tests + pot test fixes 🧪

* feat: more jug tests 🧪

* fix: code review changes 🤝

* fix: ci 👏

* fix: move rpow to its own file ✍️

* fix: rpow naming ✍️

* fix: assertEvent and naming 🔨

* fix: events checks 🔨

* feat: checkFile 🔨

EvolveArt committed:
Code Synchronisation (#43)

* feat: code sync action ✨

* Update tests.yml

* Update sync.ts

* Update sync.yml

* feat: sync tests 🔨

EvolveArt committed:
fix: remove only on tests 🧪

maciejka commented:
What is wrong with CI? @EvolveArt

EvolveArt committed:
Chore/0.10.3 (#42)

* chore: migrate cairo 0.10.2 ⬆️

* fix: python version ✍️

* chore: 0.10.3 deploy update ⬆️

- add `deployAccount` method that uses predeployed accounts on devnet
- update `simpleDeployL2`

* Update hardhat.config.ts

* fix: name ✍️

EvolveArt committed:
feat: add events checks ✨ (#46)

EvolveArt committed:
fix: disable broken tests

EvolveArt committed:
Update end.ts

EvolveArt committed:
chore: hardhat plugin 0.7.1 ⬆️

EvolveArt committed:
Update pyproject.toml

EvolveArt committed:
Update pyproject.toml

EvolveArt committed:
Delete poetry.lock

EvolveArt committed:
fix: cure tests 🧪

EvolveArt committed:
Create poetry.lock

EvolveArt committed:
fix: trace syntax ✍️ (#48)

EvolveArt committed:
fix: event handling 🔨

EvolveArt committed:
fix: remove .only

maciejka committed:
simplyfy paths

EvolveArt committed:
chore: bump dependencies ⬆️

maciejka committed:
add dai, vat views

EvolveArt committed:
fix: jug constructor params typo

maciejka committed:
add functions to safe_math

maciejka committed:
add assert

maciejka committed:
update assertions

EvolveArt committed:
Apply latest changes to end ✨ (#51)

* feat: apply latest changes to end ✨

* fix: max 🔨

EvolveArt committed:
feat: claim token ✨ (#55)

* feat: claim token ✨

* fix: end overflow max value ✍️

* fix: overflow value
